### ISSUE TITLE
Forcing protocol (static, analytic, data-driven)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/somax#23]
## Summary

Define the forcing interface for somax models. Forcing is explicit and first-class on the model (`model.forcing`), with a unified `__call__(t, grid)` signature.

Parent: Sub-epic: Core API Framework

## Design

```python
class ForcingProtocol(eqx.Module):
    """All forcing implements __call__(t, grid) -> ForcingFields."""
    def __call__(self, t: float, grid) -> ForcingFields: ...

class ConstantWindForcing(eqx.Module):
    tau_x: Float[Array, "Ny Nx"]
    tau_y: Float[Array, "Ny Nx"]
    def __call__(self, t, grid):
        return ForcingFields(tau_x=self.tau_x, tau_y=self.tau_y)

class SeasonalWindForcing(eqx.Module):
    tau0: Float[Array, ""]          # learnable amplitude!
    period: float = eqx.field(static=True)
    def __call__(self, t, grid):
        modulation = jnp.cos(2 * jnp.pi * t / self.period)
        return ForcingFields(tau_x=self.tau0 * modulation * wind_profile(grid), ...)

class InterpolatedForcing(eqx.Module):
    path: dfx.AbstractPath   # diffrax LinearInterpolation or CubicInterpolation
    def __call__(self, t, grid):
        return self.path.evaluate(t)
    @staticmethod
    def from_data(times, fields):
        return InterpolatedForcing(path=dfx.LinearInterpolation(ts=times, ys=fields))
```

## Key Design Decisions

- Forcing is explicit and first-class on the model (`model.forcing`)
- `__call__(t, grid)` signature — time-dependent by default
- Learnable params (e.g., `tau0`) are JAX arrays so `jax.grad` flows through
- Data-driven forcing uses diffrax `AbstractPath` (differentiable interpolation)
- finitevolX #154 provides raw operator primitives; somax composes them into these modules

## Package Location

`src/somax/_src/forcing/`

## Acceptance Criteria

- [ ] ForcingProtocol / ForcingFields defined
- [ ] ConstantWindForcing, SeasonalWindForcing, InterpolatedForcing implemented
- [ ] `jax.grad` flows through learnable forcing params
- [ ] InterpolatedForcing.from_data works with time series arrays
- [ ] Tests for time-dependent evaluation and gradient flow
